# Python `datetime` を利用して基準日から何日目の一覧を作る

例えば 2018 年 6 月 1 日を 1 日目とする、一連の何日目リストを出力するコードを次に示す：

In [1]:
import datetime

oneday = datetime.timedelta(days=1)
origin = datetime.date(2018, 6, 1) - oneday

def nth_date(origin, n):
    return origin + oneday * n

milestone = range(50, 501, 50)

print('旅に出てから')
for i in milestone:
    print(f'{i:>3} 日目が {nth_date(origin, i)}')

旅に出てから
 50 日目が 2018-07-20
100 日目が 2018-09-08
150 日目が 2018-10-28
200 日目が 2018-12-17
250 日目が 2019-02-05
300 日目が 2019-03-27
350 日目が 2019-05-16
400 日目が 2019-07-05
450 日目が 2019-08-24
500 日目が 2019-10-13


以下、プログラミングノート。
* 25, 50, 100, 150, ... のように間隔が一定でない場合はループの反復要素に `range` を使わず、
  何日目が欲しいのかを表す `tuple` オブジェクトを使う。
* 結局断念した `date.strftime()` で日本語の日付文字列を出力する方法がわからない。
  `locale` 絡みなので真面目に追求しても利が薄そう。
* 逆に「今日がある基準日から何日目というタイプの記念日である」ことを判定するには次のようにする。
  ログインボーナスの実装などに使いたい：

In [2]:
(datetime.date.today() - origin).days in milestone

False